src/v/http/client.cc

```diff
@@ -76,10 +77,11 @@ ss::future<client::request_response_t> client::make_request(
     auto verb = header.method();
     auto target = header.target();
     ss::sstring target_str(target.data(), target.size());
+    auto redacted = redacted_header(header);
     auto req = ss::make_shared<request_stream>(this, std::move(header));
     auto res = ss::make_shared<response_stream>(this, verb, target_str);
     prefix_logger ctxlog(http_log, ssx::sformat("[{}]", target_str));
-    vlog(ctxlog.trace, "client.make_request {}", header);
+    vlog(ctxlog.trace, "client.make_request {}", redacted);
```

Comment:
can't we just move it prior to moving and thus avoid copying in the common case of trace logging turned off?

Reply:
Good point, will try it out.